sheerwater_benchmarking/utils/caching.py

```diff
@@ -563,30 +590,22 @@ def wrapper(*args, **kwargs):
                     comp_arg_values = {key: cache_arg_values[key] for key in common_keys}
                     d = {key: d[key] for key in common_keys}
 
-                    # If they match then disable caching
-                    if comp_arg_values == d:
-                        print(f"Caching disabled for arg values {d}")
-                        cache = False
-                        break
+                    # Compare common keys
```

Comment:
@adkinsjd , this is the main logic change - I'd like to make sure you feel i'm doing this right.